ts/editor/NoteEditor.svelte

```diff
@@ -392,57 +387,29 @@ License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
     import { wrapInternal } from "@tslib/wrap";
     import LabelButton from "components/LabelButton.svelte";
     import Shortcut from "components/Shortcut.svelte";
-    import { setupImageOcclusion } from "image-occlusion";
+    import ImageOcclusionPage from "image-occlusion/ImageOcclusionPage.svelte";
     import type { IOMode } from "image-occlusion/lib";
     import { exportShapesToClozeDeletions } from "image-occlusion/shapes/to-cloze";
+    import { isShowMaskEditor } from "image-occlusion/store";
 
     import { mathjaxConfig } from "../editable/mathjax-element";
     import CollapseLabel from "./CollapseLabel.svelte";
     import * as oldEditorAdapter from "./old-editor-adapter";
 
-    $: isShowMaskEditor = false;
-    $: isIOImageLoaded = false;
-    function toggleMaskEditor(show: boolean) {
-        isShowMaskEditor = show;
-        const element = document.getElementById("io-mask-editor") as HTMLElement;
-        const ioMaskBtn = document.getElementById("io-mask-btn") as HTMLElement;
-        if (isShowMaskEditor) {
-            ioMaskBtn?.classList.add("active-io-btn");
-        } else {
-            ioMaskBtn?.classList.remove("active-io-btn");
-        }
-        if (element) {
-            element.style.display = show ? "block" : "none";
-        }
-    }
-
+    let isIOImageLoaded = false;
+    const showMaskEditor = isShowMaskEditor;
+    let imageOcclusionMode: IOMode;
```

Comment:
You later test for `if (imageOcclusionMode ...)`, so the type should be `IOMode | undefined`.